utils/global-state-update-gen/src/generic.rs

```diff
@@ -129,16 +130,25 @@ fn update_auction_state<T: StateReader>(
         );
 
         state.remove_withdraws(&validators_diff.removed);
-    }
 
-    // All entries in the new snapshot contain the same set of validators, just use the first entry.
-    new_snapshot
-        .values()
-        .next()
-        .expect("snapshot should have at least one entry")
-        .keys()
-        .cloned()
-        .collect()
+        if validators_diff.added.is_empty() && validators_diff.removed.is_empty() {
```

Comment:
I don't think that's the correct condition. It's possible to twiddle the stakes of the validators without adding or removing any of them - and in such a case that should still be `Some` (especially when we add the weights to the picture).
I think this should say something along the lines of `accounts.iter().all(|account| account.validator.is_none())`.

Reply:
Yes, that's correct. I'll add a new commit which also adds weights and will change this condition as well.